<launch>
    <env name="LD_PRELOAD" value="/usr/lib/x86_64-linux-gnu/libjemalloc.so.2"/>
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find omniveyor_mobility)/resources/cfg/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find omniveyor_mobility)/resources/cfg/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find omniveyor_mobility)/resources/cfg/local_costmap_params.yaml"  command="load" />
        <rosparam file="$(find omniveyor_mobility)/resources/cfg/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find omniveyor_mobility)/resources/cfg/move_base_params.yaml" command="load" />

        <rosparam file="$(find omniveyor_mobility)/resources/cfg/teb_local_planner_params.yaml" command="load" />
        <rosparam file="$(find omniveyor_mobility)/resources/cfg/dwa_local_planner_params.yaml" command="load" />
        <rosparam file="$(find omniveyor_mobility)/resources/cfg/carrot_planner_params.yaml" command="load" />
        <!--rosparam file="$(find omniveyor_mobility)/resources/cfg/global_planner_params.yaml" command="load" /-->

        <param name="base_global_planner" value="navfn/NavfnROS" />
        <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />

        <!-- Rotate in place is not allowed. Handled through self adaptation. -->
        <param name="clearing_rotation_allowed" value="false" />
    </node>
</launch>
